### PR TITLE
qt: Add support for building for iOS.

### PR DESCRIPTION
**Pull request by Steveice10**
Makes some changes necessary for the Qt front-end to compile for iOS.

**Note that, at least in its current form, this is not meant to be a means for official iOS support.** There are a bunch of deficiencies in the Qt front-end on iOS currently, such as the settings window going mostly off-screen (on iPhone at least, might be better on iPad) and a lack of on-screen controls. iOS support would likely need its own dedicated front-end more suited for the platform's characteristics similar to Android.

However this was trivial enough to get working that I felt it would be good to upstream to provide a means of dev testing in the interim and generally improve the portability of our Qt front-end. Plus this led to some enhancements for other platforms as well, such as improved support for generating the Info.plist metadata on macOS and additional CMake configuration options for all builds.

Do also note that this is not very useful without Vulkan merged in on top as otherwise you're limited to software rendering.

On a Mac you can use the following command to configure CMake:
```
cmake .. -DCMAKE_SYSTEM_NAME=iOS -DCMAKE_OSX_ARCHITECTURES=arm64 -DCMAKE_OSX_DEPLOYMENT_TARGET=14.0 -G Xcode
```

Then, you can open the generated project in Xcode, configure a team for code signing, and deploy to a test device.

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6594)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -15,24 +15,36 @@ project(citra LANGUAGES C CXX ASM)
...
+        set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE BOTH)
+        set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE BOTH)
+    endif()
 endif()
 
 option(ENABLE_LTO "Enable link time optimization" OFF)
 
-option(ENABLE_SDL2 "Enable the SDL2 frontend" ON)
+option(ENABLE_SDL2 "Enable using SDL2" ON)
+CMAKE_DEPENDENT_OPTION(ENABLE_SDL2_FRONTEND "Enable the SDL2 frontend" ON "NOT ANDROID AND NOT IOS" OFF)
```
Comment (SachinVin) [inline CMakeLists.txt:29]: should also depend on `ENABLE_SDL2`, and maybe even throw an error if it is enabled without `ENABLE_SDL`

File: src/common/common_paths.h [hunk 2]
```diff
@@ -21,7 +21,12 @@
 #ifdef _WIN32
 #define EMU_DATA_DIR "Citra"
 #elif defined(__APPLE__)
+#include <TargetConditionals.h>
+#if TARGET_OS_IPHONE
+#define MACOS_EMU_DATA_DIR "Documents" DIR_SEP "Citra"
```
Comment (SachinVin) [inline src/common/common_paths.h:26]: Perhaps consider renaming it to APPLE_EMU_DATA_DIR

**Review by FearlessTobi — COMMENTED** [review]
I have taken a look and don't see anything wrong with these changes, but my CMake knowledge is very limited.
Thanks for adding this!

**Comment by Steveice10** [comment]
Made two more additional fixes I identified for screen size. One is to provide a launch screen for the app so the geometry is setup properly, the other is to allow it to use iPad screen sizes instead of acting like a scaled up iPhone app.

**Comment by ansinerd** [comment]
Not a dev but ios and android enthusiast. I am genuinely so happy to see Citra finally giving attention towards iOS port. For long time it was a perception that Citra dev teams dislike iOS and would never port it to iOS but it seems things are changing probably cause of iOS being slightly less restrictive and other workarounds being available like Sideloading and JIT enabling. 
Also, i have seen Emufan helping the Antique code guy for his unofficial n3ds (citra) port with Vulkan and all. 
finally there is unofficial citra port available to iOS users, its functional but not too optimised right now, as OP rightly said,
There is not proper front end even with official port, it randomly crashes, unavailability of proper on screen controls (although physical controllers do work) , non availability of save and load state, but even then its playable in current state, on m1 chip iPad pro i am able to run mario kart 7 at 8x resolution at 30+ fps. Constant 60fps st 5/6x resolution. 


I am sure when Citra officially starts supporting iOS with proper frontend, proper optimisation of on screen controllers, proper save and load state and other settings and upstreamed, it would be unbelievably good. 

Looking forward to further development. 

Once again thank you so much

**Comment by Leakyhandle** [comment]
If you make a buy me a coffee or a Patreon I would sub to you if it helps development for iOS

**Comment by Leakyhandle** [comment]
Do y'all think we're close to making an ipa for citra if so hmu on twitter 

My Twitter is dumbmail761

**Comment by Steveice10** [comment]
Let's please keep this PR on topic to code review, thanks.

**Pull request merged**